snap/pack/pack.go

```diff
@@ -129,7 +129,11 @@ func loadAndValidate(sourceDir string) (*snap.Info, error) {
 		// TODO:UC20: optionally pass model
 		// TODO:UC20: pass validation constraints which indicate intent
 		//            to have data encrypted
-		if err := gadget.Validate(sourceDir, nil, nil); err != nil {
+		ginfo, err := gadget.ReadInfo(sourceDir, nil, &gadget.ValidationConstraints{})
```

Comment:
again is this just nil for clarity in what the arguments are? could we just use nil instead?